src/daft-sql/src/planner.rs

```diff
@@ -1860,23 +1865,23 @@ impl<'a> SQLPlanner<'a> {
             // ---------------------------------
             // signed integer
             // ---------------------------------
-            SQLDataType::Int(_) | SQLDataType::Integer(_) => DataType::Int32,
-            SQLDataType::Int2(_) | SQLDataType::SmallInt(_) => DataType::Int16,
-            SQLDataType::Int4(_) | SQLDataType::MediumInt(_) => DataType::Int32,
-            SQLDataType::Int8(_) | SQLDataType::BigInt(_) => DataType::Int64,
             SQLDataType::TinyInt(_) => DataType::Int8,
+            SQLDataType::Int2(_) | SQLDataType::SmallInt(_) | SQLDataType::Int16 => DataType::Int16,
+            SQLDataType::Int(_) | SQLDataType::Integer(_) | SQLDataType::Int4(_) | SQLDataType::MediumInt(_)  | SQLDataType::Int32  => DataType::Int32,
+            SQLDataType::Int8(_) | SQLDataType::BigInt(_) | SQLDataType::Int64 => DataType::Int64,
+
             // ---------------------------------
             // unsigned integer
             // ---------------------------------
-            SQLDataType::UnsignedInt(_) | SQLDataType::UnsignedInteger(_) => DataType::UInt32,
-            SQLDataType::UnsignedInt2(_) | SQLDataType::UnsignedSmallInt(_) => DataType::UInt16,
-            SQLDataType::UnsignedInt4(_) | SQLDataType::UnsignedMediumInt(_) => DataType::UInt32,
-            SQLDataType::UnsignedInt8(_) | SQLDataType::UnsignedBigInt(_) => DataType::UInt64,
+
+            SQLDataType::UnsignedInt2(_) | SQLDataType::UnsignedSmallInt(_)| SQLDataType::UInt16 => DataType::UInt16,
+            SQLDataType::UnsignedInt4(_) | SQLDataType::UnsignedMediumInt(_) | SQLDataType::UnsignedInt(_) | SQLDataType::UnsignedInteger(_) | SQLDataType::UInt32 => DataType::UInt32,
```

Comment:
Drop MediumInt and UnsignedMedium INT. It's non-standard and we already have three aliases.